scality_sproxyd_client/sproxyd_client.py

```diff
@@ -76,6 +81,11 @@ def __init__(self, sproxyd_urls, conn_timeout=None, proxy_timeout=None,
         self.sproxyd_urls_set = set()
 
         for sproxyd_url in sproxyd_urls:
+            if re.match(r'^http(s)?://', sproxyd_url) is None:
```

Comment:
You could use `urlparse.urlparse`, and then assert the result's `method` attribute is in `['http', 'https']`. This adds some more validation as well:

```python
>>> urlparse.urlparse('http://[:::::1:81/chord')
...
ValueError: Invalid IPv6 URL
```